pkg/scheduler/partition.go

```diff
@@ -1208,3 +1219,11 @@ func (pc *PartitionContext) removeAllocationAsk(appID string, allocationKey stri
 		}
 	}
 }
+
+func (pc *PartitionContext) cleanupExpiredApps() {
```

Comment:
The `PartitionManager` is a different go routine. That means the delete below can cause a race condition.
We either need to lock around the delete or make the function locked. I think speed wise we should just lock around the delete.